### PR TITLE
Add travis-ci configuration

### PR DESCRIPTION
**Pull request by jgoldfar**
Not sure if there's any interest in enabling something like this, but it does build correctly! Thanks for your work on SWI-Prolog

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - berkeley-db
+      - pcre
+      - jpeg
+
+git:
+  submodules: false
+
+before_script:
+  - mkdir build
+  - if [ "$TRAVIS_OS_NAME" == "linux" ] ; then sed -i 's~url = ..~url = https://github.com/SWI-Prolog~' .gitmodules; fi
```
Comment (JanWielemaker) [inline .travis.yml]: Why is this needed?   Seems hacky and possibly this can be avoided at a different level?

Comment (jgoldfar) [reply .travis.yml]: I agree, and I admit I didn’t look for the most elegant way to get this particular part working. Having a sub module configuration with relative paths means that in order to run a CI service one has to have all of the corresponding git repositories cloned to your username on GitHub. This isn’t an issue for your (upstream) repository, but anyone else (e.g. me, to set this up) would be unable to get things running. I could remove or comment out this line.

Comment (JanWielemaker) [reply .travis.yml]: At least I now understand the why :)  If you only want to deal with the core you can also simply not clone the packages.  That should work fine if you use `-DINSTALL_DOCUMENTATION=OFF`. I propose to include the file above, commenting the git hackery.  Agree?  I'm happy to do so (will apply and amend the commit).

**Comment by jgoldfar** [comment]
That all sounds good to me. Thanks

**Comment by JanWielemaker** [comment]
Merged after update as agreed.  Please create a new pull request relative to the current master if you want further changes.

**Comment by JanWielemaker** [comment]
This pull request has been mentioned on **SWI-Prolog**. There might be relevant details there:

https://swi-prolog.discourse.group/t/consolidating-the-71-github-repositories-to-simplify-maintenance-and-contribution/492/3

**Comment by JanWielemaker** [comment]
This pull request has been mentioned on **SWI-Prolog**. There might be relevant details there:

https://swi-prolog.discourse.group/t/being-friendly-to-quick-contributions/493/1
